### PR TITLE
Chore: Add CI to check FP tests for every pull request

### PR DESCRIPTION
**Pull request by bournezjc**
# Summary
This pr adds a CI for every pull request of this repo to run FP tests

# Test plan
https://github.com/babylonlabs-io/finality-gadget/actions/runs/11743557577
image

### CODE REVIEW

**Comment by bournezjc** [comment]
The commit head of this pr looks wierd here. 
image
Any idea where does this "2082471076753073b7ec445a151d5af2b5c67029" commit hash come from?

**Review by SebastianElvis — DISMISSED** [review]
Nice!

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -25,4 +28,34 @@ jobs:
...
+          current_hash=$(git rev-parse HEAD)
+          remote_url=$(git remote get-url origin)
+          if [[ $remote_url == https://* ]]; then
+              # For HTTPS URLs
+              repo_path=$(echo $remote_url | sed -E 's|https://github.com/(.+)(\.git)*|\1|')
+          else
+              # For SSH URLs
+              repo_path=$(echo $remote_url | sed -E 's/.*:(.+)\.git/\1/')
+          fi
+          fp_version=$(go list -m github.com/$repo_path@$current_hash)
```
Comment (SebastianElvis) [inline .github/workflows/publish.yml:50]: The goal here seems to get the version of the current FP repository. Can we achieve this by `git rev-parse HEAD`?

Comment (bournezjc) [reply .github/workflows/publish.yml:50]: > The goal here seems to get the version of the current FP repository. Can we achieve this by `git rev-parse HEAD`?

line 41. However it seems only works for push action. For pull request, should I use HEAD~ instead?

Comment (bournezjc) [reply .github/workflows/publish.yml:52]: Looks good by now  :)
https://github.com/babylonlabs-io/finality-gadget/actions/runs/11772186929/job/32787208311

**Review by SebastianElvis — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -25,4 +28,36 @@ jobs:
...
+          git clone https://github.com/babylonlabs-io/finality-provider
+          cd finality-provider
+          git checkout base/consumer-chain-support
+          sed -i "s|github\.com\/babylonlabs-io\/finality-gadget.*|$fp_version|g" go.mod
+          go mod tidy
+
+          export PATH=${PATH}:$(go env GOPATH)/bin
+
+          make op-e2e-devnet
+          make test-e2e-op
```
Comment (lesterli) [inline .github/workflows/publish.yml:63]: it is better to split these into new steps: update fp version and run op e2e test

**Review by lesterli — APPROVED** [review]

**Pull request merged**